### PR TITLE
Obg 213 mock list

### PR DESCRIPTION
**Pull request by electronicpeter**
list accounts and list tx runs with mocks

### CODE REVIEW

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/mapper/ManualMapper.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import de.adorsys.opba.fintech.api.model.generated.AccountList;
+import de.adorsys.opba.fintech.api.model.generated.BankDescriptor;
+import de.adorsys.opba.fintech.api.model.generated.BankProfile;
+import de.adorsys.opba.fintech.api.model.generated.TransactionsResponse;
+import de.adorsys.opba.tpp.banksearch.api.model.generated.BankProfileDescriptor;
+import org.mapstruct.factory.Mappers;
+
+@SuppressWarnings("checkstyle:HideUtilityClassConstructor")
+public class ManualMapper {
```
Comment (DG0lden) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/mapper/ManualMapper.java]: What is the purpose of this class?

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java [hunk 2]
```diff
@@ -2,47 +2,49 @@
...
 @Slf4j
 @RequiredArgsConstructor
 public class AccountService {
+    @Value("${mock.tppais.listaccounts}")
+    String mockTppAisString;
 
     private final TppAisClient tppAisClient;
 
-    public InlineResponse2003 listAccounts(ContextInformation contextInformation, UserEntity userEntity, String bankId) {
+    public InlineResponse2003 listAccounts(ContextInformation contextInformation, SessionEntity sessionEntity, String bankId) {
```
Comment (DG0lden) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java]: same recommendation: provide named object in OpenAPI

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java [hunk 3]
```diff
@@ -2,47 +2,49 @@
...
+        if (mockTppAisString != null && mockTppAisString.equalsIgnoreCase("true") ? true : false) {
```
Comment (DG0lden) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java]: How about using Apache `BooleanUtils.toBoolean(mockTppAisString)`?

File: fintech-examples/fintech-ui/proxy-dev.conf.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "/fintech-api-proxy/*": {
```
Comment (DG0lden) [inline fintech-examples/fintech-ui/proxy-dev.conf.json]: we probably need to bind it to nginx environment variable to run it in docker

File: fintech-examples/fintech-ui/src/app/dashboard/home/home.component.html [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<div class="row mt-5">
+<div class="row mt-5" *ngIf="cardList2 && cardList2.length">
```
Comment (DG0lden) [inline fintech-examples/fintech-ui/src/app/dashboard/home/home.component.html]: `ngIf="cardList2` shall be enough, it's javascript

File: fintech-examples/fintech-impl/pom.xml [hunk 6]
```diff
@@ -41,6 +47,11 @@
             <artifactId>spring-cloud-starter-openfeign</artifactId>
             <version>2.2.1.RELEASE</version>
         </dependency>
+        <dependency>
+            <groupId>io.github.openfeign</groupId>
+            <artifactId>feign-annotation-error-decoder</artifactId>
+            <version>1.3.0</version>
```
[the remark this answers is not in the document]
Comment (electronicpeter) [reply fintech-examples/fintech-impl/pom.xml]: removed

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/controller/FinTechAccountInformationImpl.java [hunk 7]
```diff
@@ -34,9 +40,22 @@
...
         }
         ContextInformation contextInformation = new ContextInformation(xRequestID);
-        UserEntity userEntity = userRepository.findByXsrfToken(xsrfToken).get();
+        SessionEntity sessionEntity = userRepository.findByXsrfToken(xsrfToken).get();
 
-        return new ResponseEntity<>(accountService.listAccounts(contextInformation, userEntity, bankId), HttpStatus.OK);
+        return new ResponseEntity<>(accountService.listAccounts(contextInformation, sessionEntity, bankId), HttpStatus.OK);
+    }
+
+    public ResponseEntity<InlineResponse2004> aisTransactionsGET(String bankId, String accountId, UUID xRequestID,
```
[the remark this answers is not in the document]
Comment (electronicpeter) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/controller/FinTechAccountInformationImpl.java]: This api was done by francis. I will do interfaces changes in #240

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/SessionEntity.java [hunk 8]
```diff
@@ -37,7 +40,22 @@
...
+            @AttributeOverride(name = "okURL", column = @Column(name = "TRANS_OK_URL")),
+            @AttributeOverride(name = "notOkURL", column = @Column(name = "TRANS_NOT_OK_URL"))
+    })
+    RedirectUrlsEmbeddable redirectListTransactions;
+
+    @AttributeOverrides({
+            @AttributeOverride(name = "okURL", column = @Column(name = "ACCOUNTS_OK_URL")),
+            @AttributeOverride(name = "notOkURL", column = @Column(name = "ACCOUNTS_NOT_OK_URL"))
+    })
+    @Column(name = "b")
```
[the remark this answers is not in the document]
Comment (electronicpeter) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/SessionEntity.java]: as explained, no more valid code. already completly removed in #240

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/SessionEntity.java [hunk 9]
```diff
@@ -37,7 +40,22 @@
     @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = false)
     private List<CookieEntity> cookies = new ArrayList<>();
 
-    public UserEntity addCookie(String key, String value) {
+    @AttributeOverrides({
+            @AttributeOverride(name = "okURL", column = @Column(name = "TRANS_OK_URL")),
+            @AttributeOverride(name = "notOkURL", column = @Column(name = "TRANS_NOT_OK_URL"))
+    })
+    RedirectUrlsEmbeddable redirectListTransactions;
```
[the remark this answers is not in the document]
Comment (electronicpeter) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/database/entities/SessionEntity.java]: as explained, no more valid code. already completly removed in #240

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/mapper/ManualMapper.java (hunk 1 above)
Comment (electronicpeter) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/mapper/ManualMapper.java]: To gather all automatic wrappers in one place. So I dont have to inject all of them.

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java (hunk 2 above)
Comment (electronicpeter) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java]: yes I can do with 240

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java (hunk 3 above)
Comment (electronicpeter) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java]: didnt know that. thx

Comment (electronicpeter) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java]: didnt know that. thx

File: fintech-examples/fintech-ui/proxy-dev.conf.json (hunk 4 above)
Comment (electronicpeter) [reply fintech-examples/fintech-ui/proxy-dev.conf.json]: this was done by sascha. It came in due to rebase.

File: fintech-examples/fintech-ui/src/app/dashboard/home/home.component.html (hunk 5 above)
Comment (electronicpeter) [reply fintech-examples/fintech-ui/src/app/dashboard/home/home.component.html]: this was done by sascha. It came in due to rebase.

**Pull request merged**